homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -149,6 +149,54 @@ def put(self, request, username):
         )
 
 
+class HueFullStateView(HomeAssistantView):
+    """Return full state view of emulated hue."""
+
+    url = '/api/{username}'
+    name = 'emulated_hue:username:lights:state'
+    extra_urls = ['/api/{username}/']
+    requires_auth = False
+
+    def __init__(self, config):
+        """Initialize the instance of the view."""
+        self.config = config
+
+    @core.callback
+    def get(self, request, username):
+        """Process a request to get the list of available lights."""
+        if not is_local(request[KEY_REAL_IP]):
+            return self.json_message('only local IPs allowed',
+                                     HTTP_BAD_REQUEST)
+
+        json_response = [{
+            'error': {
+                'address': '/',
+                'description': 'unauthorized user',
+                'type': '1'
+                }
+            }]
+
+        if username == '12345678901234567890':
```

Comment:
What is going on here ?

Reply:
Well, since you wanted to remove the view for /api/null and /api/(null), I had to make a if to separate /api/{username} if no valid username was used (the app does not continue it stalls, it has not done the initial “handshake”).

As I think/tried I’ve explained a couples of times now. Apps using hue (same for the app “sleep on android”) first send an “ping/hello”-something with a false username for example “(null)”. If it gets the response that’s it is not a correct username it will use the implementation already done (before this pull ), otherwise it will just wait forever. After that it is possible to collect the lamps. 

But I can create an custom component instead, I cant remove the function and still have the functionality for init left.

Reply:
Every device sends a different username. So you can't just hardcode to some username. 

I think the proper solution here is that emulated hue needs to keep track of authorized usernames. That will create the right experience that Sleep Cycle expects.